packages/@lwc/template-compiler/src/parser/expression.ts

```diff
@@ -91,64 +95,56 @@ function validateSourceIsParsedExpression(source: string, parsedExpression: Node
     ]);
 }
 
-export function parseExpression(source: string, config: ResolvedConfig): TemplateExpression {
-    try {
-        const parsed = parseExpressionAt(source, 1, { ecmaVersion: 2020 });
-
-        validateSourceIsParsedExpression(source, parsed);
-        validateExpression(parsed, config);
-
-        return parsed;
-    } catch (err) {
-        err.message = `Invalid expression ${source} - ${err.message}`;
-        throw err;
-    }
+export function parseExpression(
+    source: string,
+    ctx: ParserCtx,
+    location: Location
+): TemplateExpression | never {
+    return ctx.withErrorWrapping(
+        () => {
+            const parsed = parseExpressionAt(source, 1, { ecmaVersion: 2020 });
+
+            validateSourceIsParsedExpression(source, parsed);
+            validateExpression(parsed, ctx.config);
+
+            return parsed;
+        },
+        ParserDiagnostics.TEMPLATE_EXPRESSION_PARSING_ERROR,
+        location,
+        (err) => `Invalid expression ${source} - ${err.message}`
+    );
 }
 
-export function parseIdentifier(source: string): TemplateIdentifier | never {
+export function parseIdentifier(
```

Comment:
I would recommend passing the context as the first argument. This will match the function signature in `index.ts`.